### PR TITLE
Interpolate pev garch

### PR DESCRIPTION
**Pull request by juniperlsimonis**
this branch updates the pois_env_ts model to include data interpolation and to also include a no-covariate (i.e. intercept only) model
the model is now renamed to pevGARCH (this is a Poisson environmental variate GARCH model)

associated with the model update, there is also an update to the get_weather_data function to produce the weather data associated with each new moon (even if there was no census, in order to match the interpolated abundances values) as well.

### CODE REVIEW

**Review by ethanwhite — CHANGES_REQUESTED** [review]
Looks great overall. The one major thing is that the idea behind all of the lag code is unclear. Some comments to explain what it's doing would be useful. I also made a couple of minor cleanup suggestions.

Hopefully we can get these added today so this can be merged in before tomorrow's forecast build.

File: models/pevgarch.R [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    species <- colnames(abundances)[2:(ncol(abundances) - 3)]
+
+    interpolated_abundances <- interpolate_abundance(abundances)
+
+    fit_start_newmoon <- min(abundances$newmoonnumber)
+    fit_end_newmoon <- max(abundances$newmoonnumber)
+    initial_newmoon <- max(abundances$newmoonnumber)
+
+    zero_abund_mean <- rep(0, 12)
+    zero_abund_interval <- matrix(0, ncol = 2, nrow = 12)
```
[the remark this answers is not in the document]
Comment (juniperlsimonis) [reply models/pevgarch.R]: ooh good catch on that! 
this actually also is tied with the argument to the model function named `num_forecast_newmoons` which has been hardcoded as 12 in all the models, but should be based on the length of `forecast_months` that comes out of the metadata
i'll address this here, and will go back to the other models to tidy that as well.

**Comment by juniperlsimonis** [comment]
I just added a bit of text re: the weather lag to the header in the model script. Let me know if you think that's sufficient or if I should add in an explanation of the lag elsewhere.

File: tools/model_functions.R [hunk 2]
```diff
@@ -127,13 +131,29 @@ get_weather_data <- function(moons, all, lag){
...
+  which_is_lagged = rep(NA, nrow(weather_data))
+  for(i in 1:nrow(weather_data)){
+    specific <- which(weather_data$newmoonnumber == weather_data$NewMoonNumber_with_lag[i])
+    if(length(specific) > 0){
+      which_is_lagged[i] = specific
+    }
+  }
+  weather_data$year_with_lag = weather_data$year[which_is_lagged]
+  weather_data$month_with_lag = weather_data$month[which_is_lagged]
+
```
Comment (gmyenni) [inline tools/model_functions.R:149]: This 'which_is_lagged' bit could use an explanation at the top

File: tools/model_functions.R [hunk 3]
```diff
@@ -127,13 +131,29 @@ get_weather_data <- function(moons, all, lag){
...
   weather_data = weather_data %>%
     select(-newmoondate, -censusdate, -period, -year, -month) %>%
-    right_join(all, by=c('NewMoonNumber_with_lag'='newmoonnumber')) %>%
-    select(year,month,mintemp,maxtemp,meantemp,precipitation,ndvi,newmoonnumber,NewMoonNumber_with_lag)
-  
+    right_join(all_moons_table, by=c('NewMoonNumber_with_lag'='newmoonnumber')) %>%
+    select(year_with_lag,month_with_lag,mintemp,maxtemp,meantemp,precipitation,ndvi,newmoonnumber,NewMoonNumber_with_lag)
+ 
+  colnames(weather_data)[1:2] <- c("year", "month")
+
```
Comment (gmyenni) [inline tools/model_functions.R:162]: Yep, looks good to me! I don't know if there's much you can do to make all this rearranging more readable. We could have some context at the top, like 'a 6-month lag black boxes the entire rain - resources - rodents process'. But the lag won't always be 6. . .

**Review by gmyenni — CHANGES_REQUESTED** [review]

**Comment by gmyenni** [comment]
Okay, you could add the biological context I suggested above in the model header. 
And in model_functions.R you can add a reference back to the model use at the start of the lag section.

**Comment by gmyenni** [comment]
I guess 'above' isn't really that helpful. :) I was saying you could add something like 'a 6-month lag black boxes the entire rain - resources - rodents process' to the model script header.

Overall though, I think this looks good. There are overarching issues with the lag stuff that is making the code unavoidably confusing and weird that we'll have to get to later.

**Review by gmyenni — APPROVED** [review]

**Review by ethanwhite — APPROVED** [review]

**Comment by juniperlsimonis** [comment]
I was having a difficult time figuring out how to resolve that conflict, as it wasn't letting me do it on the web interface and the instructions it gave me weren't working in the command line.
it looks like the conflict had to do with the removal of the old model file, so I added it back here for the merge and then will delete it in a new quick branch and merge off the new master. Note that in the quick interim, the build and check might break because of the change in the weather data.

**Pull request merged**